wgpu/src/lib.rs

```diff
@@ -100,15 +101,51 @@ pub use wgt::{
     QUERY_SIZE, VERTEX_STRIDE_ALIGNMENT,
 };
 
-#[cfg(not(webgpu))]
-#[doc(hidden)]
-pub use ::hal;
-#[cfg(feature = "naga")]
-pub use ::naga;
-#[cfg(not(webgpu))]
-#[doc(hidden)]
+/// Re-export of our `wgpu-core` dependency.
+///
+#[cfg(wgpu_core)]
+#[doc(inline)]
 pub use ::wgc as core;
 
+/// Re-export of our `wgpu-hal` dependency.
+///
+///
+#[cfg(wgpu_core)]
+#[doc(inline)]
+pub use ::hal;
+
+/// Re-export of our `naga` dependency.
+///
+#[cfg(wgpu_core)]
+#[cfg_attr(
+    docsrs,
+    doc(cfg(any(wgpu_core, feature = "glsl", feature = "spirv", feature = "naga-ir",)))
```

Comment:
why not use `any(wgpu_core, naga)`

Reply:
Basically there's no good intuition on what "naga" represents to the outside user. wgpu_core is also a bit weird, but it's at least more clear.